### PR TITLE
cxx-qt-gen: signals as extern block

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Related to #557

### CODE REVIEW

**Comment by ahayzen-kdab** [comment]
This doesn't remove the emit wrapper yet (even though it's pointless now). I'll rebase the other branch #556 on top of this one to do the tidy ups

File: crates/cxx-qt-gen/src/parser/cxxqtdata.rs [hunk 1]
```diff
@@ -200,14 +194,24 @@ impl ParsedCxxQtData {
     }
 
     fn try_parse_inherit_verbatim(&mut self, tokens: TokenStream) -> Result<Option<Item>> {
```
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-gen/src/parser/cxxqtdata.rs:196]: Needs to be renamed to `try_parse_verbatim`, as it's no longer limited to `inherit`.

File: book/src/getting-started/1-qobjects-in-rust.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ Then you can use the afformentioned features with the help of more macros.
 - `#[cxx_qt::qobject]` - Expose a Rust struct to Qt as a QObject subclass.
     - `#[qproperty]` - Expose a field of the Rust struct to QML/C++ as a [`Q_PROPERTY`](https://doc.qt.io/qt-6/qtqml-cppintegration-exposecppattributes.html#exposing-properties).
     - `#[qinvokable]` - Expose a function on the QObject to QML and C++ as a [`Q_INVOKABLE`](https://doc.qt.io/qt-6/qtqml-cppintegration-exposecppattributes.html#exposing-methods-including-qt-slots).
-- `#[cxx_qt::qsignals(T)]` - Use an enum to define the [Signals](https://doc.qt.io/qt-6/signalsandslots.html#signals) of a QObject T.
+- `#[cxx_qt::qsignals]` - Use an enum to define the [Signals](https://doc.qt.io/qt-6/signalsandslots.html#signals) of a QObject T.
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply book/src/getting-started/1-qobjects-in-rust.md:50]: done

File: crates/cxx-qt-gen/test_inputs/passthrough_and_naming.rs [hunk 3]
```diff
@@ -91,9 +91,9 @@ pub mod ffi {
         }
     }
 
-    #[cxx_qt::qsignals(MyObject)]
-    pub enum MySignals {
-        Ready,
+    #[cxx_qt::qsignals]
+    unsafe extern "C++" {
+        fn ready(self: Pin<&mut qobject::MyObject>);
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/test_inputs/passthrough_and_naming.rs:96]: done

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Pull request merged**